File: keevo/backend/src/main/java/com/keevo/catalog/product/domain/entity/ProductStatus.java
package com.keevo.catalog.product.domain.entity;

/**
 * Product status enumeration
 * 
 * ACTIVE - Product is active and visible in POS
 * DRAFT - Product created on-the-fly during POS sale (Story 2.4)
 */
public enum ProductStatus {
    ACTIVE,
    DRAFT
}

File: keevo/app/lib/features/catalog/domain/model/product_status.dart
/// ProductStatus — lifecycle status of a product.
///
/// - [active]: fully created, visible in POS catalogue
/// - [draft]: created on-the-fly during a POS sale (Story 2.4)
enum ProductStatus {
  active,
  draft;

  /// Convert to backend string value.
  String get value => name.toUpperCase();

  /// Parse from backend JSON string.
  static ProductStatus fromString(String s) =>
      ProductStatus.values.firstWhere(
        (e) => e.value == s.toUpperCase(),
        orElse: () => ProductStatus.active,
      );
}
